### PR TITLE
Development

### PR DESCRIPTION
**Pull request by kelvinwijaya**
# Description

### V1.3.0-SNAPSHOT
+ Update test-suites framework to cover more use-case scenarios
+ Add Utility to read from PKI Key with PEM format
+ Update README documentation

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [x] Code enhancement and update (non-breaking change)
- [ ] Security patch

## How Has This Been Tested?
Junit and new in build test-framework

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -140,13 +140,53 @@ dependencies {
...
+String realm = "<<your_client_host_url>>"
+String authPrefix = "<<authPrefix>>
+String httpMethod = "get"
+String url = "https://<<Target_URL>>/api/v1/?param1=first&param2=123";
+String certFileName = "certificates/ssc.alpha.example.com.p12";
+String password = "<<passphrase>>";
+String alias = "alpha";
+String appId = "<<appId>>";
+String secret = null;
+ApiList formList = null;
```
Comment (robincher) [inline README.md]: Need to indicate formList is optional and also describe it?

File: README.md [hunk 2]
```diff
@@ -156,7 +196,7 @@ Params:
 * timestamp - set to null for current timestamp
 
 ```java
-String url = "https://<<URL>>/api/v1/?param1=first&ab-param2=123";
+String url = "https://<<Target_URL>>/api/v1/?param1=first&param2=123";
```
Comment (robincher) [inline README.md]: Need to share that if they put queryParams formList, they no need to put the query param in the URL ?

File: src/main/java/com/api/util/ApiSecurity/ApiSigning.java [hunk 3]
```diff
@@ -353,7 +362,50 @@ public static PublicKey getPublicKeyFromX509Certificate(String publicCertificate
         log.debug("Exit :: getPublicKeyFromX509Certificate");
         return pk;
     }
-
+    
+    /**
+     * @param key
+     * @return
+     * @throws IOException
```
Comment (robincher) [inline src/main/java/com/api/util/ApiSecurity/ApiSigning.java]: The javadoc comment lack a description what this method do?

File: src/main/java/com/api/util/ApiSecurity/ApiSigning.java [hunk 4]
```diff
@@ -564,4 +628,36 @@ private static String getNewNonce() throws NoSuchAlgorithmException {
         return nonce;
     }
 
+
```
Comment (robincher) [inline src/main/java/com/api/util/ApiSecurity/ApiSigning.java]: Add javadocs

File: README.md (hunk 1 above)
Comment (kelvinwijaya) [reply README.md]: updated Readme.md

File: src/main/java/com/api/util/ApiSecurity/ApiSigning.java (hunk 3 above)
Comment (kelvinwijaya) [reply src/main/java/com/api/util/ApiSecurity/ApiSigning.java]: updated javadoc

File: src/main/java/com/api/util/ApiSecurity/ApiSigning.java (hunk 4 above)
Comment (kelvinwijaya) [reply src/main/java/com/api/util/ApiSecurity/ApiSigning.java]: Updated javadoc

**Comment by kelvinwijaya** [comment]
merge to master for version 1.3.0

**Pull request merged**